CharityPortal/api/api.go

```diff
@@ -6,61 +6,87 @@ import (
+	provider, err := oidc.NewProvider(ctx, providerURL)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot create oidc provider; %w", err)
+	}
+
+	oidcConfig := &oidc.Config{ClientID: cfg.ClientID}
+	verifier := provider.Verifier(oidcConfig)
+
+	oauth2Config := &oauth2.Config{
+		ClientID:     cfg.ClientID,
+		ClientSecret: cfg.ClientSecret,
+		RedirectURL:  cfg.RedirectURL,
+		Endpoint:     provider.Endpoint(),
+		Scopes:       []string{oidc.ScopeOpenID, "profile", "email"},
+	}
+	return oauth2Config, verifier, nil
+}
+
+const environmentDevelopment = "development"
 
 func NewAPI(cfg *config.Config) (*API, error) {
 	setupLogger(cfg.Logger)
-	authProvider, err := setupAuthProvider(cfg, cfg.Environment)
+	oauth2Config, tokenVerifier, err := configureAuth(*cfg.Auth)
 	if err != nil {
 		return nil, err
 	}
-	router := newRouter(authProvider)
+	sessionManager := auth.NewSessionManager(cfg.Auth.HashKey, cfg.Auth.BlockKey, oauth2Config)
+	var protect func(next http.HandlerFunc) http.HandlerFunc
+	if cfg.Environment == environmentDevelopment {
+		protect = middlewares.ProtectFake
+	} else {
+		protect = middlewares.BuildProtect(sessionManager.ReadSession)
+	}
 
+	router := newRouter(tokenVerifier, sessionManager, protect)
 	server := http.Server{
 		Addr:    ":8080",
 		Handler: router,
 	}
-
+	slog.Info("started http://localhost:8080")
 	return &API{
 		server: &server,
 	}, nil
 }
 
-func newRouter(authProvider auth.AuthProvider) *http.ServeMux {
+func newRouter( 
+	tokenVerifier *oidc.IDTokenVerifier,
+	sessionManager *auth.SessionManager,
+	protect func(next http.HandlerFunc) http.HandlerFunc,
+	) http.Handler {
 	mux := http.NewServeMux()
 
-	commonMiddlewares := alice.New(middlewares.Log, middlewares.NewSessionMiddleware(authProvider).Session)
-	notLoggedOnlyMiddlewares := commonMiddlewares.Extend(alice.New(middlewares.NewAuthMiddleware(authProvider).NotLogged))
-	loggedOnlyMiddleware := commonMiddlewares.Extend(alice.New(middlewares.NewAuthMiddleware(authProvider).LoggedOnly))
 	dataConfirmationService := dataconfirmation.NewDataConfirmationService()
-
 	fs := http.FileServer(http.Dir("./web/static"))
-	mux.Handle("GET /static/", commonMiddlewares.Then(http.StripPrefix("/static/", fs)))
-
-	mux.Handle("GET /", notLoggedOnlyMiddlewares.Then(handlers.NewHomeHandler()))
-	mux.Handle("POST /login", notLoggedOnlyMiddlewares.Then(handlers.NewLoginHandler(authProvider)))
-	mux.Handle("GET /login/callback", notLoggedOnlyMiddlewares.Then(handlers.NewLoginCallbackHandler(authProvider)))
-	mux.Handle("POST /logout", loggedOnlyMiddleware.Then(handlers.NewLogoutHandler()))
-
-	mux.Handle("GET /dashboard", loggedOnlyMiddleware.Then(handlers.NewDashboardHandler()))
-	mux.Handle("POST /data-confirmation", loggedOnlyMiddleware.Then(handlers.NewDataConfirmationHandler(dataConfirmationService)))
-	return mux
+	mux.Handle("GET /static/", http.StripPrefix("/static/",fs))
+	mux.Handle("GET /", handlers.NewHomeHandler())
```

Comment:
Weird formatting here and in rest of the files.

Reply:
Actually vscode formats the stuff for me. What can I make "unweird"?

Reply:
Use neovim

Reply:
enforcing formatting by using the same IDE sounds like "buy DIESEL car so my gas station won't have to serve gasolin".

Reply:
I've runned go fmt on whole solution and configured VsCode to format on save. I guess this will work.

Reply:
Good that VScode has such option.